flag.go

```diff
@@ -21,12 +21,18 @@ var (
 	commaWhitespace = regexp.MustCompile("[, ]+.*")
 )
 
-// BashCompletionFlag enables bash-completion for all commands and subcommands
-var BashCompletionFlag = &BoolFlag{
-	Name:   "generate-bash-completion",
+// GenerateCompletionFlag enables completion for all commands and subcommands
+var GenerateCompletionFlag = &BoolFlag{
+	Name:   "generate-completion",
 	Hidden: true,
 }
 
+// InitCompletionFlag generates completion code
+var InitCompletionFlag = &StringFlag{
+	Name:  "init-completion",
+	Usage: "generate completion code. Value must be 'bash' or 'zsh'",
+}
```

Comment:
We can consider the current completion code as default completion code. Another flag could then be added later to provide custom completion code.